tools/cold-backup/run.sh

```diff
@@ -0,0 +1,481 @@
+            kubectl -n $NAMESPACE scale $eo_deploy --replicas 0
+            __wait_for delete pod strimzi.io/name=$cluster_name-entity-operator
+        fi
+        local ke_deploy="$(kubectl -n $NAMESPACE get deploy $cluster_name-kafka-exporter -o name --ignore-not-found)"
+        if [[ -n $ke_deploy ]]; then
+            kubectl -n $NAMESPACE scale $ke_deploy --replicas 0
+            __wait_for delete pod strimzi.io/name=$cluster_name-kafka-exporter
+        fi
+        local kafka_ss="$(kubectl -n $NAMESPACE get statefulset $cluster_name-kafka -o name --ignore-not-found)"
+        if [[ -n $kafka_ss ]]; then
+            kubectl -n $NAMESPACE scale $kafka_ss --replicas 0
+            __wait_for delete pod strimzi.io/name=$cluster_name-kafka
+        fi
+        local zoo_ss="$(kubectl -n $NAMESPACE get statefulset $cluster_name-zookeeper -o name --ignore-not-found)"
+        if [[ -n $zoo_ss ]]; then
+            kubectl -n $NAMESPACE scale $zoo_ss --replicas 0
+            __wait_for delete pod strimzi.io/name=$cluster_name-zookeeper
+        fi
+    fi
+}
+
+__cleanup() {
+    if [[ $INCREMENTAL == false ]]; then
+        rm -rf $__TMP/$NAMESPACE/$CLUSTER_NAME
+    fi
+    kubectl -n $NAMESPACE delete pod $RSYNC_POD_NAME 2>/dev/null ||true
+    __start_cluster $CLUSTER_NAME
+}
+
+__error() {
+    local message="${1-"Error"}"
+    echo "$message"
+    exit 1
+}
+
+__confirm() {
+    read -p "Please confirm (y/n) " reply
+    if [[ ! $reply =~ ^[Yy]$ ]]; then
+        echo "Exiting"
+        exit 0
+    fi
+}
+
+__wait_for() {
+    local condition="$1"
+    local type="$2"
+    local label="$3"
+    if [[ -n $condition && -n $type && -n $label ]]; then
+        local resource=$(kubectl -n $NAMESPACE get $type -l $label -o name --ignore-not-found)
+        if [[ -n $resource ]]; then
+            echo "Waiting for $condition of $type with label $label"
+            kubectl -n $NAMESPACE wait $type -l $label --for="$condition" --timeout="${WAIT_TIMEOUT_SEC}s"
+        fi
+    else
+        __error "Missing required parameters"
+    fi
+}
+
+__export_env() {
+    local cluster_name="$1"
+    local dest_path="$2"
+    if [[ -n $cluster_name && -n $dest_path ]]; then
+        echo "Exporting environment data"
+        local filter="data-$cluster_name-zookeeper"
+        ZK_REPLICAS=$(kubectl -n $NAMESPACE get kafka $cluster_name -o yaml | yq eval ".spec.zookeeper.replicas" -)
+        ZK_PVC_SIZE=$(kubectl -n $NAMESPACE get pvc $filter-0 -o yaml | yq eval ".spec.resources.requests.storage" -)
+        ZK_PVC_CLASS=$(kubectl -n $NAMESPACE get pvc $filter-0 -o yaml | yq eval ".spec.storageClassName" -)
+        local kafka_storage_type=$(kubectl -n $NAMESPACE get kafka $cluster_name -o yaml | yq eval ".spec.kafka.storage.type" -)
+        if [[ $kafka_storage_type == "jbod" ]]; then
+            JBOD_VOL_NUM=$(kubectl -n $NAMESPACE get pvc | grep $cluster_name-kafka-0 | wc -l)
+            filter="data-0-$cluster_name-kafka"
+        else
```

Comment:
How does this handle when the JBOD storage is using more then one volume? Also, keep in mind that the volume IDs are customizable. So users might for example have JBOD volumes with IDs 3, 5 and 8. (I did not tried JBOD with multiple volumes, so maybe it is handled fine ... this just made me wonder if it does)

Reply:
OK